### PR TITLE
BytesStreamExtension add stream based xml decoding

### PR DESCRIPTION
**Pull request by Leptopoda**
Leftover from my investigation for #1085
I don't expect any significant performance gains from this. Some quick testing revealed a 3% gain at most which I consider margin of error.

### CODE REVIEW

File: packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart [hunk 1]
```diff
@@ -76,7 +76,7 @@ class DynamiteRawResponse<B, H> {
         _rawBody = switch (bodyType) {
           const FullType(Uint8List) => await response.bytes,
           const FullType(String) => await response.string,
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart:78]: Please also rename those to `bytesBody` and `stringBody`

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart:78]: I think we should rather drop the `Body` suffix from xml and json.
The extension methods are on `Stream<List<int>>` and therefore have nothing to do with the response itself.

What do you think?

Comment (provokateurin) [reply packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart:78]: Sounds good

File: packages/dynamite/dynamite_runtime/lib/src/http_extensions.dart [hunk 2]
```diff
@@ -3,13 +3,18 @@ import 'dart:convert';
...
 /// A stream of bytes.
 ///
 /// Usually a `Stream<Uint8List>`.
 typedef BytesStream = Stream<List<int>>;
 
-final _utf8JsonDecoder = utf8.decoder.fuse(json.decoder);
+final _xmlConverter =
+    utf8.decoder.fuse(XmlEventDecoder()).fuse(const XmlNormalizeEvents()).fuse(const XmlNodeDecoder());
+
+final _jsonBytesConverter = utf8.decoder.fuse(json.decoder);
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/lib/src/http_extensions.dart:17]: The names are not following the same scheme, is that intended?

**Review by provokateurin — APPROVED** [review]

**Pull request merged**